### PR TITLE
Tracing

### PR DESCRIPTION
**Pull request by lukaszPulawski**

### CODE REVIEW

File: core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java [hunk 1]
```diff
@@ -15,16 +15,52 @@
...
+        this.callListeners = callListeners;
+    }
+
+    public void addCallListeners(RemoteServiceCallListener... callListeners) {
+        for (RemoteServiceCallListener cl : callListeners) {
+            this.callListeners.add(cl);
+        }
+    }
+
+    @SuppressWarnings("PMD")
```
Comment (pgrela) [inline core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java]: Warum gibt es eine allgemeine PMD?

File: core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java [hunk 2]
```diff
@@ -15,16 +15,52 @@
...
 import org.springframework.remoting.RemoteConnectFailureException;
 import org.springframework.remoting.caucho.HessianProxyFactoryBean;
 
+import com.payu.ratel.context.ProcessContext;
+import com.payu.ratel.context.RemoteServiceCallEvent;
+import com.payu.ratel.context.RemoteServiceResponseEvent;
+import com.payu.ratel.context.ServiceCallInput;
+import com.payu.ratel.context.ServiceCallResult;
+
 public class RatelHessianProxyFactoryBean extends HessianProxyFactoryBean {
```
Comment (pgrela) [inline core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java]: Have you considered using decorator pattern instead of inheritance?

Comment (lukaszPulawski) [inline core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java]: I suggest to use interceptors for this purpsoe. I have written this case explicitly to Issue #45.

File: core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java (hunk 1 above)
Comment (lukaszPulawski) [inline core/src/main/java/com/payu/ratel/client/RatelHessianProxyFactoryBean.java]: external API enforces catching java.lang.Exception

**Pull request merged**